main.py

```diff
@@ -141,14 +143,44 @@ def login(driver, username, password):
     email_input = driver.find_element_by_xpath("//input[@name='emailAddress']")
     email_input.clear()
     email_input.send_keys(username)
+    
     password_input = driver.find_element_by_xpath("//input[@name='password']")
     password_input.clear()
     password_input.send_keys(password)
 
     LOGGER.info("Logging in")
     driver.find_element_by_xpath("//input[@value='SIGN IN']").click()
-    wait_until_visible(driver=driver, xpath="//a[@data-path='myAccount:greeting']")
+    
+    sleep_val = 3
+    num_retries_attempted = 0
+    num_retries = 5
+    while True:
+        try:
+            time.sleep(sleep_val)
+            xpath = "/html/body/div[2]/div[3]/div[3]/div/div[2]/input"
+            wait_until_visible(driver=driver, xpath=xpath, duration=5)
+            driver.find_element_by_xpath(xpath).click()
+        
+            password_input = driver.find_element_by_xpath("//input[@name='password']")
+            password_input.clear()
+            password_input.send_keys(password)
+
+            LOGGER.info("Logging in")
+            driver.find_element_by_xpath("//input[@value='SIGN IN']").click()
+            sleep_val += random.randint(0,10)
+
+            if num_retries_attempted < num_retries:
+                num_retries_attempted += 1
+                continue
+            else:
+                LOGGER.info("Too many login attempts. Please restart app.")
+                break
+        except Exception as e:
+            LOGGER.exception("Error dialog did not load, proceed. Error: " + str(e))
```

Comment:
What if an exception occurs during `driver.find_element_by_xpath("//input[@value='SIGN IN']").click()`? You'd never increment `num_retries_attempted`.

Reply:
If that scenario occurs, the loop should break as I added a break line after that error exception. I can test to double check this.

Reply:
We could do that, but I don't know what the Nike server will do if we retry to login too many times, that's why I limited the retries to 5. I didn't dare try with my account and ip :) 

I suppose I can add a catch exception for the sign-in click and if it fails, and increment the `num_retries_attempted` in the catch